src/olympia/devhub/decorators.py

```diff
@@ -10,9 +10,27 @@
 from olympia.addons.decorators import addon_view_factory
 from olympia.addons.models import Addon
 from olympia.amo.decorators import login_required
+from olympia.users.models import SuppressedEmail
 from olympia.versions.models import Version
 
 
+def suppressed_email(f):
+    @functools.wraps(f)
+    def wrapper(request, *args, **kwargs):
+        suppressed_email = False
+
+        if request.user.is_authenticated:
+            suppressed_email = SuppressedEmail.objects.filter(
+                email=request.user.email
+            ).exists()
+
+        print('suppressed_email decorator', suppressed_email)
+        request.suppressed_email = suppressed_email
```

Comment:
For new code I would prefer we avoid custom properties on the `request` object, especially if it's not something that needs to exist for every request.

Since ultimately for our purposes what we care about is whether or not an user has their email suppressed... I suggest something along those lines:
- Remove the decorator completely
- Introduce a `is_email_suppressed()` or similarly named method in `UserProfile` that does `return SuppressedEmail.objects.filter(email=self.email).exists()`
- In `suppressed_email.html`, check with `if waffle.switch('suppressed-email') and request.user.is_authenticated and request.user.has_email_suppressed()`

Reply:
Sounds good. This would be a great thing to have a decision log for. I initially implemented with extending request as I had seen several instances of that pattern in the code. If it's something we want to move away from it would be great to document that decision.